### PR TITLE
feat: Add a comment for a product

### PR DESCRIPTION
**Pull request by Timmy-id**
## What does the PR do?

This PR lets a logged-in user write a comment under a product

How should this be manually tested?

 - The logged-in user makes POST request to the `/api/v1/organizations/:orgId/products/:productId/comments` endpoint with the following payload:

- The request for the endpoint:

#### HEADER
``` Authorization: Bearer <token> ```

#### BODY
```json
{
  "comment":  string,
}
```

#### Successful Response

Upon successful product update, the API should return a 201 CREATED status code.
**Status Code: 201**

```json
    {
      status: "success",
      message: 'Comment added successfully'',
      status_code: 201,
      data: {
           "id": "frfr3435",
           "product_id": "tyu3-434ld-le3",
           "comment": "New comment",
           "user_id": "ghkjuj-23nju2-lm1",
           "created_at": "2024-08-05T22:20:05.360Z"
      }
    }
```
product-comment

#### Validation Error:
**Status Code: 400**

```json
{
     "message": [
           "comment should not be empty"
       ],
       "error": "Bad Request",
       "statusCode": 400
}
```
bad-request-comment

#### Authorization Error:
**Status Code: 401**

```json
{
    "message": "User is currently unauthorized, kindly authenticate to continue",
    "status": 401
}
```
unauthorized-comment

#### Not Found Error
**Status Code: 404**

```json
{
    "error": "Product not found",
    "status_code": 404
}
```
not-found-comment

Task

- [x] Create a POST request to ```/api/v1/organizations/:orgId/products/:productId/comments``` to create a comment under a product.
- [x] Implement server-side logic to handle successful and failed update requests.
- [x] Ensure that only logged-in users can add a comment.
- [x] Each request to the endpoint included a valid authentication token
- [x] Validate that the comment field is present and is a string.

### CODE REVIEW

File: src/modules/products/products.service.ts [hunk 1]
```diff
@@ -202,4 +206,41 @@ export class ProductsService {
...
     };
   }
+
+  async addCommentToProduct(productId: string, commentDto: AddCommentDto, userId: string) {
+    const { comment } = commentDto;
+    const product = await this.productRepository.findOne({ where: { id: productId } });
+    const user = await this.userRepository.findOne({ where: { id: userId } });
+
+    if (!product) {
+      throw new NotFoundException({
```
Comment (Cobidex) [inline src/modules/products/products.service.ts:216]: Use customHttpExceptions provided in the helper directory. reference auth service for guide on usage

File: src/modules/products/products.service.ts [hunk 2]
```diff
@@ -202,4 +206,41 @@ export class ProductsService {
...
+        error: 'Product not found',
+        status_code: HttpStatus.NOT_FOUND,
+      });
+    }
+
+    if (!user) {
```
Comment (Cobidex) [inline src/modules/products/products.service.ts:222]: use custom exception filter, and its better to use Constants for error and success messages where possible

Comment (Cobidex) [inline src/modules/products/products.service.ts:222]: Also if this is a protected endpoint, why still checking if user is present.

File: src/modules/products/products.service.ts (hunk 1 above)
Comment (Timmy-id) [reply src/modules/products/products.service.ts:216]: Done

File: src/modules/products/products.service.ts (hunk 2 above)
Comment (Timmy-id) [reply src/modules/products/products.service.ts:222]: Fixed

**Review by CFCIfe — COMMENTED** [review]
LGTM

**Review by Heba-WebDev — APPROVED** [review]

**Review by Cobidex — APPROVED** [review]

**Pull request merged**